Toplean_Alexandru/week-02/Domain/Domain/GetMenusOp/GetMenusOp.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Domain.Models;
+using Infrastructure.Free;
+using LanguageExt;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using static Domain.Domain.GetMenusOp.GetMenusResult;
+
+namespace Domain.Domain.GetMenusOp
+{
+    internal class GetMenusOp : OpInterpreter<GetMenusCmd, IGetMenusResult, Unit>
+    {
+        public override Task<IGetMenusResult> Work(GetMenusCmd Op, Unit state)
+        {
+            if (Exists(Op.Restaurant))
+            {
+                if (Op.IsValid())
+                {
+                    // Menus are successfully selected
+                    return Task.FromResult<IGetMenusResult>(new MenusGot(Op.Menus));
+                }
+                else
+                {
+                    // No menus were selected, Menu List of the restaurant was null
+                    return Task.FromResult<IGetMenusResult>(new NoMenusGot($"Restaurant {Op.Restaurant.Name} has no menus"));
```

Comment:
This is where you should implemented the retrieval of Menus. This is where the logic should reside, instead of the GetMenusCmd constructor.